### PR TITLE
Late Day Exceptions UI

### PR DESCRIPTION
**Pull request by pbailie**
NEW FILE:    TAGradingServer/account/admin-latedays-exceptions.php

Feature ready for code review to merge to master

AREAS TO IMPROVE
* Still no scrollbar when student table gets clipped at the bottom
* When CSV validation fails, a general error message is issued.
- Maybe expand error messages to explain what went wrong during validation
* Double check proposed/disabled SQL for "new schema" as I was unable to test

### CODE REVIEW

File: TAGradingServer/account/admin-latedays-exceptions.php [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+	COUNT(1)
+FROM users
+WHERE
+	user_id=?
+AND
+	user_group=0
+SQL;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ END DISABLED CODE */
+
+	\lib\Database::beginTransaction();
```
Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: Remove this line and the one for commit. There's no reason to wrap a single SQL statement (especially if it's a select) in a transaction as there's no way to leave the database in a broken state.

File: TAGradingServer/account/admin-latedays-exceptions.php [hunk 2]
```diff
@@ -0,0 +1,752 @@
...
+SELECT
+	g_id
+FROM
+	gradeables
+WHERE
+	g_title=?
+SQL;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ END DISABLED CODE */
+
+	\lib\Database::beginTransaction();
```
Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: Same as above

File: TAGradingServer/account/admin-latedays-exceptions.php [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+	$sql = <<<SQL
+SELECT
+	MAX(g_id)
+FROM gradeables
+WHERE
+	g_gradeable_type=0"
+SQL;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ END DISABLED CODE */
+
+	\lib\Database::beginTransaction();
```
Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: same as above

File: TAGradingServer/account/admin-latedays-exceptions.php [hunk 4]
```diff
@@ -0,0 +1,752 @@
...
+	g_id,
+	g_title
+FROM gradeables
+WHERE
+	g_gradeable_type=0"
+ORDER BY g_id DESC;
+SQL;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ END DISABLED CODE */
+
+	\lib\Database::beginTransaction();
```
Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: same as above

File: TAGradingServer/account/admin-latedays-exceptions.php [hunk 5]
```diff
@@ -0,0 +1,752 @@
...
+	users.user_group=0
+AND
+	late_day_exceptions.late_day-exceptions IS NOT NULL
+AND
+	late_day_exceptions.late_day_excpetions>0
+ORDER BY users.user_email ASC;
+SQL;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ END DISABLED CODE */
+
+	\lib\Database::beginTransaction();
```
Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: same as above

File: TAGradingServer/account/admin-latedays-exceptions.php [hunk 6]
```diff
@@ -0,0 +1,752 @@
+<?php
+
+//Author: Peter Bailie, Systems Programmer, RPI Computer Science, July 2016
+
+/* MAIN ===================================================================== */
+
+include "../header.php";
+
+check_administrator();
```
Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: This function causes the script to die if the administrator check fails. The if statement is extraneous and the else branch will never be executed (as the script would die)

Comment (pbailie) [inline TAGradingServer/account/admin-latedays-exceptions.php]: I can remove the ```if ($user_is_administrator)``` condition, but it does exist in a few other "admin" php pages.

Comment (MasterOdin) [inline TAGradingServer/account/admin-latedays-exceptions.php]: Yeah, it's something that should be removed from all pages most likely, but not worth the effort at this point. You don't really need to change anything here, I'm just stating that the else branch is impossible to execute (so there's no way to test it).

**Pull request merged**